From 2ea2e19661a81ac729f0b299c7e948c855dcb8b5 Mon Sep 17 00:00:00 2001
From: Sridharan S N <quic_sridsn@quicinc.com>
Date: Tue, 21 Nov 2023 12:04:17 +0530
Subject: [PATCH 452/500] mtd: spi-nor: ensure flash operations are complete
 before shutdown

Ensure there are no pending flash operation when system halts
Reference commit:864de914888f4c700ee24028183e21de8fe259b7

Change-Id: Id1c3cefc43c9788c4b510cd4265b40a5747eb02f
Signed-off-by: Sridharan S N <quic_sridsn@quicinc.com>
---
 drivers/mtd/spi-nor/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/mtd/spi-nor/core.c b/drivers/mtd/spi-nor/core.c
index 0fa53954f220..a5eed7bf0406 100644
--- a/drivers/mtd/spi-nor/core.c
+++ b/drivers/mtd/spi-nor/core.c
@@ -3712,6 +3712,9 @@ static void spi_nor_shutdown(struct spi_mem *spimem)
 {
 	struct spi_nor *nor = spi_mem_get_drvdata(spimem);
 
+	/* Ensure no pending flash operation in progress */
+	spi_nor_wait_till_ready(nor);
+
 	spi_nor_restore(nor);
 }
 
-- 
2.34.1

